bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Image.java

```diff
@@ -2256,15 +2258,20 @@ public void setBackground(Color color) {
 	device.internal_dispose_GC(hDC, null);
 }
 
-private void setCurrentDeviceZoom(int newZoomFactor) {
-	if (this.currentDeviceZoom != newZoomFactor) {
-		this.currentDeviceZoom = newZoomFactor;
+private void setCurrentNativeZoom(int newZoom) {
+	int oldZoom = this.getZoom();
+	if (oldZoom != newZoom) {
```

Comment:
The naming is now confusing, because `oldZoom` referes to a "corrected" zoom value, whereas `newZoom` is a native zoom value. Maybe `newZoom` should be called `newNativeZoom` instead? In addition, the comparison `oldZoom != new[Native]Zoom` probably requires a comment to understand why native and non-native zoom values are compared.

Reply:
I have made changes in this method to fix the names and usage. However, in a bigger picture it is a stale method and would be removed in the image PR. The comparision should be between old device zoom and new device zoom, because we dont wanna create many handles for the same device zoom. (Since multiple native zooms can have same device zoom)

Reply:
Two further remarks on this:
1. Even if a method is supposed to be removed in the future, current changes to it still need to be sound and of good quality. No one can guarantee that a currently intended subsequent change will happen, so we can usually only review and decide based on what exists and what is done in this particular PR.
2. I understand the goal of having only one handle per device zoom, but it still looks wrong that an image does not update its native zoom just because the actually used zoom calculated from the native zoom did not change. In my opinion, the necessity not to update something when the native zoom changed without affecting the actual zoom has to be done in some recalculation logic but not when setting the native zoom value.

Reply:
I have resolved this now. Also made sure that it wont have any side effects with generatign new handles.